agent/api/ecsclient/client.go

```diff
@@ -419,6 +425,25 @@ func trimString(inputString string, maxLen int) string {
 	}
 }
 
+func (client *APIECSClient) buildManagedAgentStateChangePayload(change api.ManagedAgentStateChange) *ecs.ManagedAgentStateChange {
```

Comment:
nit: this function looks prettier (more readable) like so:

```golang
func (client *APIECSClient) buildManagedAgentStateChangePayload(change api.ManagedAgentStateChange) *ecs.ManagedAgentStateChange {
	if !change.Status.ShouldReportToBackend() {
		seelog.Warnf("Not submitting unsupported managed agent state %s for container %s in task %s",
			status.String(), change.ContainerName, change.TaskArn)
		return nil
	}
	var trimmedReason *string
	if change.Reason != "" {
		trimmedReason := aws.String(trimString(change.Reason, ecsMaxReasonLength))
	}
	return &ecs.ManagedAgentStateChange{
		ManagedAgentName: aws.String(change.Name),
		ContainerName:    aws.String(change.ContainerName),
		Status:           aws.String(change.Status.String()),
		Reason:           trimmedReason,
	}
}
```

Reply:
changed.